subt/main.py

```diff
@@ -454,10 +454,10 @@ def on_pose2d(self, timestamp, data):
         self.xyz = x, y, z
         self.trace.update_trace(self.xyz)
         # pose3d
-        dist3d = quaternion.rotate_vector([dist, 0, 0], self.orientation)
-        self.xyz_quat = [a + b for a, b in zip(self.xyz_quat, dist3d)]
-        xyz_quat = [p + o for p, o in zip(self.xyz_quat, self.offset)]
-        self.bus.publish('pose3d', [xyz_quat, self.orientation])
+        if self.xyz_quat is not None and self.orientation is not None:
+            dist3d = quaternion.rotate_vector([dist, 0, 0], self.orientation)
+            self.xyz_quat = [a + b for a, b in zip(self.xyz_quat, dist3d)]
```

Comment:
I must say, that even I am already confused by `xyz_quat` name :( ... yes, it was `xyz` (i.e. 3D) integrated over `quat`ernions.

Reply:
Suggestions for a better name welcome.